dot/digest/digest.go

```diff
@@ -225,6 +257,35 @@ func (h *Handler) handleBlockFinalisation(ctx context.Context) {
 	}
 }
 
+// setBABEDigestsOnFinalization is called only when a block is finalised
+// and defines the correct next epoch data and next config data.
+func (h *Handler) setBABEDigestsOnFinalization(finalizedHeader *types.Header) error {
```

Comment:
We are setting next epoch data and next config data here. Rename the function signature to reflect that. `setBABEDigestsOnFinalization` seems misleading to me.


tbh, I liked the separate functions as they were previously,

Reply:
Can we change the name to `persistBABEDigestsForNextEpoch` and keep this function as it is?

Reply:
> @kishansagathiya can we change the name to persistBABEDigestsForNextEpoch and keep this function as it is?

why?
that's basically same as what you have right now.

An appropriate name would be setNextEpochDataAndSetNextConfigData, And then the error that you get could be because of not being able to set epoch data or config data. But a better would be to just split this function into setNextEpochData and setNextConfigData. `setNextEpochDataAndSetNextConfigData` looks like two separate funcs are merged together.

If addressing this makes your life difficult (having to change a lot of code, not being able to write tests), I am cool with you not splitting the function.